inference-engine/src/mkldnn_plugin/nodes/mkldnn_reorder_node.cpp

```diff
@@ -213,15 +245,17 @@ void MKLDNNReorderNode::optimizedNspc2Ncsp() {
     auto src_data = reinterpret_cast<const float *>(parentEdge->getMemoryPtr()->GetPtr());
     auto dst_data = reinterpret_cast<float *>(childEdge->getMemoryPtr()->GetPtr());
 
-    const size_t stride1 = DIM2 * DIM3 * DIM4;
-    const size_t stride0 = stride1 * DIM1;
-    parallel_for2d(DIM0, stride1, [&](size_t b, size_t j) {
-        auto src_off = b*stride0 + j*DIM1;
-        auto dst_off = b*stride0 + j;
+    const auto dstStrides = childEdge->getMemoryPtr()->GetDescWithType<BlockedMemoryDesc>().getStrides();
+    const size_t block_size = DIM2 * DIM3 * DIM4;
+    const size_t src_batch_stride = block_size * DIM1;
+    const size_t dst_batch_stride = childEdge->getChild()->isInplace() ? dstStrides[0] : src_batch_stride;
```

Comment:
shouldn't it work correct for non inplace case?
same for `optimizedNcsp2Nspc`
```suggestion
    const size_t dst_batch_stride = dstStrides[0];
```

Reply:
It should, you're right. Corrected.